lib/ansible/module_utils/basic.py

```diff
@@ -242,6 +243,8 @@
     delimiter=dict(),  # used by assemble
     directory_mode=dict(),  # used by copy
     unsafe_writes=dict(type='bool'),  # should be available to any module using atomic_move
+    lock=dict(type='bool', default=False),
+    lock_timeout=dict(type='int', default=0)
```

Comment:
I would default to None and allow the 0 to be immediate, vs no timeout .. possibly -1 or any negative value for 'no timeout'